subsys/bluetooth/mesh/adv.c

```diff
@@ -202,29 +202,30 @@ struct net_buf *bt_mesh_adv_buf_get(k_timeout_t timeout)
 	return process_events(events, ARRAY_SIZE(events));
 }
 
-struct net_buf *bt_mesh_adv_buf_get_by_tag(uint8_t tag, k_timeout_t timeout)
+struct net_buf *bt_mesh_adv_buf_get_by_tag(enum bt_mesh_adv_tag_bit tags, k_timeout_t timeout)
 {
-	if (IS_ENABLED(CONFIG_BT_MESH_ADV_EXT_FRIEND_SEPARATE) && tag & BT_MESH_FRIEND_ADV) {
+	if (IS_ENABLED(CONFIG_BT_MESH_ADV_EXT_FRIEND_SEPARATE) &&
+	    tags & BT_MESH_ADV_TAG_BIT_FRIEND) {
 		return net_buf_get(&bt_mesh_friend_queue, timeout);
 	}
 
-#if CONFIG_BT_MESH_RELAY_ADV_SETS
-	if (tag & BT_MESH_RELAY_ADV) {
-		return net_buf_get(&bt_mesh_relay_queue, timeout);
+	if (tags & BT_MESH_ADV_TAG_BIT_LOCAL) {
+		return bt_mesh_adv_buf_get(timeout);
 	}
-#endif
 
-	return bt_mesh_adv_buf_get(timeout);
+#if CONFIG_BT_MESH_RELAY_ADV_SETS
```

Comment:
Hm.. How is it built if `CONFIG_BT_MESH_RELAY_ADV_SETS` is zero? Seems C compilation error.

Reply:
Will not compile.

Reply:
Got it.